bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -499,14 +518,26 @@ func (r *KubeadmConfigReconciler) joinControlplane(ctx context.Context, scope *S
 
 	verbosityFlag := ""
 	if scope.Config.Spec.Verbosity != nil {
-		verbosityFlag = fmt.Sprintf("--v %s", strconv.Itoa(int(*scope.Config.Spec.Verbosity)))
+		verbosityFlag =
+
+			fmt.Sprintf("--v %s", strconv.Itoa(int(*scope.Config.Spec.Verbosity)))
+	}
+
+	var files = append(certificates.AsFiles(), scope.Config.Spec.Files...)
+
+	additionalFiles, err := r.resolveFiles(ctx, scope.Config)
+	if err != nil {
+		scope.Error(err, "Failed to resolve files")
+		return ctrl.Result{}, err
 	}
 
+	files = append(files, additionalFiles...)
```

Comment:
I noticed these files appending logic is repeated in multiple places, can we provide a utility method somewhere to detail with it?

Reply:
how's that?